lib/spack/spack/util/gpg.py

```diff
@@ -28,15 +28,18 @@ def parse_keys_output(output):
 
 
 class Gpg(object):
+    _gpg = None
+
     @staticmethod
     def gpg():
         # TODO: Support loading up a GPG environment from a built gpg.
-        gpg = Executable('gpg2')
-        if not os.path.exists(GNUPGHOME):
-            os.makedirs(GNUPGHOME)
-            os.chmod(GNUPGHOME, 0o700)
-        gpg.add_default_env('GNUPGHOME', GNUPGHOME)
-        return gpg
+        if Gpg._gpg is None:
+            Gpg._gpg = which('gpg2', 'gpg')
+            if not os.path.exists(GNUPGHOME):
+                os.makedirs(GNUPGHOME)
+                os.chmod(GNUPGHOME, 0o700)
+            Gpg._gpg.add_default_env('GNUPGHOME', GNUPGHOME)
+        return Gpg._gpg
```

Comment:
Caching the executable like done here makes the fixture:

https://github.com/spack/spack/blob/2eadfa24e961a01b23a54d0c5fb609ae7af4f49a/lib/spack/spack/test/cmd/ci.py#L36-L41

not effective and causes multiple test keys to be created in `$spack_root/opt/spack/gpg/`